dev-util/plan9port: fix build, bump version and drop old

neeshy opened the pull request.

neeshy commented:
Despite the fact that on GitHub this PR has been assigned to @blueness, it seems that he is no longer a maintainer of the plan9port package.
@mgorny @juippis

juippis commented:
Seems like it was proxy-maintained until he was retired due to inactiveness. 

You need to split your commit into two commits - one for adding new version, one for removing. I see EAPI-5 there so please check that when you add a new version, it's up-to-date with Gentoo's modern standards. Also use repoman before committing.

neeshy commented:
It should be good to go. I've bumped the EAPI version to 7, and I've added a 99999999 version.

neeshy commented:
@juippis

neeshy commented:
@juippis @mgorny Could one of you review this PR?

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -0,0 +1,132 @@
+
+S="${WORKDIR}/${MY_P}"
+
+PLAN9="/usr/lib/plan9"
```
How does this work in a multilib system?

neeshy replied:
I don't know, I don't have a multilib system to test it on. I think it should be sufficient to use get_libdir here, or just use /opt instead.

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -0,0 +1,132 @@
+	docompress -x "${PLAN9}/man"
+
+	# do* plays with the executable bit, and we should not modify them
+	cp -a * "${ED}/${PLAN9}"
```
This is starting to look like it belongs in /opt instead...

neeshy replied:
I think so too. blueness was using /usr/lib/plan9 but the documentation in install.txt prefers /opt/plan9 anyway.

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -0,0 +1,132 @@
+		ROOTPATH="${EPLAN9}/bin"
+		MANPATH="${EPLAN9}/man"
+	EOF
+	doenvd "${T}/60plan9"
```
```
newenvd - 60plan9 <<- _EOF_
	PLAN9="${EPLAN9}"
	PATH="${EPLAN9}/bin"
	ROOTPATH="${EPLAN9}/bin"
	MANPATH="${EPLAN9}/man"
_EOF_
```

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -0,0 +1,132 @@
+	elog "versions of common UNIX tools, use the absolute path:"
+	elog "${PLAN9}/bin or the 9 command (eg: 9 troff)"
+	elog
+	elog "Please report any bugs to bugs.gentoo.org, NOT Plan9Port."
```
This looks like something you want to print on first install, not after every install. Perhaps *readme.gentoo-r1* eclass is useful here?

neeshy commented:
@juippis I've addressed your review.
I'm using readme.gentoo-r1.eclass and everything is installed in /opt now.

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -99,34 +112,24 @@ src_compile() {
-	# P9P's man does not handle compression
-	docompress -x "${PLAN9}/man"
+	# cleanup
+	rm -f LOCAL.config install.{log,sum}
```
Don't use `-f`. You need to know whether these files exist or not.

neeshy replied:
```
/var/db/repos/gentoo % rg -l -g'*.ebuild' 'rm -r?f' | wc -l
2215
```
I don't get why this is a problem.

juippis replied:
It makes users/maintainers miss errors with their ebuilds. ::gentoo is not always the best place to look at, most of the ebuilds are horrible and outdated. `-f` is only acceptable when something is not fully predictable.

juippis commented on dev-util/plan9port/plan9port-99999999.ebuild:
```diff
@@ -0,0 +1,126 @@
+# Copyright 1999-2020 Gentoo Authors
```
Why the filename is `99999999` instead of standard `9999`?

neeshy replied:
The version of the other ebuild is a date with 8 characters (20200330).

juippis replied:
> I personally use it. I think it makes sense to have one because upstream plan9port doesn't actually have set releases.

Okay, thanks for clarifying. Because this was already in tree, I thought this is how they name their releases. There is a common way to write ebuilds for programs that have no releases. You name them `program-0_pre20200330`, so when 1.0 or even 0.1 release is made, it will always be an upgrade. Seeing this, I now strongly believe this should be re-versioned such. It will cause a downgrade after this PR is merged, but after that, it works like it's supposed to. And can co-exist with -9999 ebuild.

juippis commented on dev-util/plan9port/plan9port-20200330.ebuild:
```diff
@@ -0,0 +1,132 @@
+
+DESCRIPTION="Port of many Plan 9 programs and libraries"
+HOMEPAGE="https://9fans.github.io/plan9port/"
+SRC_URI="https://github.com/9fans/${PN}/archive/${MY_HASH}.tar.gz -> ${MY_P}.tar.gz"
```
I'd prefer this to be `-> ${P}.tar.gz`, but doesn't matter much.

neeshy commented:
> Also are you sure you want to add that -9999 ebuild here? It can be added later on. I see no value in it.

I personally use it. I think it makes sense to have one because upstream plan9port doesn't actually have set releases.

neeshy commented:
Squashed and pushed.

juippis commented on dev-util/plan9port/plan9port-99999999.ebuild:
```diff
@@ -0,0 +1,129 @@
+	readme.gentoo_create_doc
+
+	# cleanup
+	rm -f LOCAL.config install.{log,sum}
```
Take off `-f` and append `|| die` here too.

juippis commented:
Yep, compiled with gcc-9. Also
```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/plan9/bin/mklatinkbd
 * /opt/plan9/bin/rio
 * /opt/plan9/bin/secstored
 * /opt/plan9/bin/secuser
 * /opt/plan9/bin/winwatch
 * /opt/plan9/bin/xshove
 * /opt/plan9/bin/yacc
```

neeshy commented:
>  I've accidentally depcleaned my gcc-9 and this doesn't compile. I guess it's due to gcc-10.

Yea, it was due to an -fno-common issue. This was addressed by: https://github.com/9fans/plan9port/commit/c1c1b5267fd5e14be531a4b22ed0124b35d427cb. I've bumped the non-9999 version of the ebuild to fix this.

> `-f` is only acceptable when something is not fully predictable.

It turns out that that install.sum and install.log are only created when running `./INSTALL -c` (which is avoided in this ebuild). I removed those from the `rm` call, but in the end I decided not to delete LOCAL.config as well, just so that all files from the build directory show up in the package.

> QA Notice: Files built without respecting LDFLAGS have been detected

I added a QA_FLAGS_IGNORED variable.

neeshy commented:
With https://github.com/9fans/plan9port/commit/9444b8e4bc847f8fd9d02466976b962288cedf31 CFLAGS/LDFLAGS can be set, so QA_FLAGS_IGNORED has correspondingly been removed.

neeshy commented:
https://github.com/9fans/plan9port/issues/395 is currently breaking builds on non-BSD systems, so I added a temporary patch for it.

neeshy commented:
> This might've changed upstream after you made the latest bump.

Looks like it was fixed by https://github.com/9fans/plan9port/commit/c53ad837a734f7570badcb3666ccb3604e7e6467. Bumping.

neeshy commented:
@juippis Is this ready to merge?

juippis commented:
Ok, it sitll didn't apply cleanly. I added these files manually, please check that what ends up in tree also works.

You should put version to patch file names.
